analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -784,31 +784,33 @@ public void Catch_NestedInFinally()
     [TestMethod]
     public void CatchWhen_Simple()
     {
-        const string code = @"
-var tag = ""BeforeTry"";
-try
-{
-    Tag(""InTry"");
-}
-catch (Exception ex) when (ex is FormatException)
-{
-    tag = ""InCatch"";
-}
-finally
-{
-    tag = ""InFinally"";
-}
-tag = ""End"";";
+        const string code = """
+            var tag = "BeforeTry";
+            try
+            {
+                Tag("InTry");
+            }
+            catch (Exception ex) when (ex is FormatException)
+            {
+                tag = "InCatch";
+            }
+            finally
+            {
+                tag = "InFinally";
+            }
+            tag = "End";
+            """;
         var validator = SETestContext.CreateCS(code).Validator;
         validator.ValidateTagOrder(
             "BeforeTry",
             "InTry",
             "InFinally",
+            "InFinally",
```

Comment:
This is the correct behavior. From `InTry` he code can either flow to `InFinally` (no exception thrown) or to `InCatch` and then to `InFinally` (exception thrown).

Reply:
There's a lot going on line 813. What about adding "because" argument to those assertions to explain this?